### ISSUE TITLE
Cannot change developer settings after TestButler.setup()

### ISSUE THREAD

**Issue by kurtzmarc** [linkedin/test-butler#78]
We change the value of the "Don't keep activities" developer setting in between some tests. We use UI-Automator to navigate the developer options screen, find the "don't keep" cell, click on the switch, and then exit the developer options. After installing test-butler I noticed that when the test changed the setting via UI automator it wasn't "sticking". I set some debug breakpoints and through manipulating the UI manually while debugging I found that before we call TestButler.setup() I can go into the developer settings, change the value, exit, return to the developer settings, and I would find the value to still be set. But after I call TestButler.setup() and go through all that, the setting will no longer stick.

Is there anything with test-butler that would "lock" those developer settings? Is there any workaround?

**Comment by Kisty** [comment]
The only way I can think of is running a suite for each config using `adb shell am...`: one with don't keep activities dev settings, one without. See AndroidJUnitRunner for script parameters https://developer.android.com/reference/android/support/test/runner/AndroidJUnitRunner. Sorry I can't be of any more help

**Comment by kurtzmarc** [comment]
Is it to be expected to not be able to change those developer settings when test-butler is initialized?

**Comment by drewhannay** [comment]
Hm, I don't know why this would be happening. TestButler definitely modifies a lot of system settings as it's being set up, but it's weird that this would prevent you from changing other settings via the UI. 

One workaround for your particular use case might be to add a feature to TestButler so that IT can change the "Don't keep activities" setting (I assume this is just another setting key that we can change the same way we do with all the others). But this doesn't really explain why the problem is happening in the first place.

If you're up to it, you could try removing some of the code in `ButlerService#onCreate()` and try to figure out precisely which part is causing your issue.

**Comment by kurtzmarc** [comment]
I particularly like adding the ability to change this setting from TestButler (much much faster than navigating the UI). It would be nice to have a sanity check from someone else that this problem exists for them too. I have debugged it to the point where I can see the behavior change once I call TestButler.setup(), however I don't know if something else we're doing contributes to the issue. If I find anything else out, I'll update the ticket.

**Comment by kurtzmarc** [comment]
I downloaded the repo and tried your test app on an emulator. I was able to replicate the issue on 5.1 and 6.0, but not in 7.0. So it appears to be platform specific. Regardless I am going to submit a PR that adds "always keep activities" support to TestButler. Hopefully we can push that through, because I'd love to have that option!

**Comment by drewhannay** [comment]
Thanks for the contribution! Unless someone wants to further investigate the original issue, I'll close this one.

**Comment by kurtzmarc** [comment]
I did a quick check and it appears that the custom activity controller is the culprit.
```
// Install custom IActivityController to prevent system dialogs from appearing if apps crash or ANR
NoDialogActivityController.install();
```
I set breakpoints in the custom controller. Everything seems to start correctly and nothing seems to be crashing. Logcat doesn't reveal anything.

**Comment by kurtzmarc** [comment]
Any chance to get a new release pushed out that includes this?

**Comment by drewhannay** [comment]
Done! Just released 1.4.0
